### DISCUSSION TITLE
There are still some issues in the Chinese display in the terminal

### DISCUSSION THREAD

**Discussion by angzicn** [pybricks/support#1922] [Q&A] [answered]
I really appreciate everything you have done before about #1742 
There are still some issues in the Chinese display in the terminal like this↓
![微信截图_20241108161309](https://github.com/user-attachments/assets/fc74397c-ca66-4d3b-b9df-8669ce829b1c)
and when I use Chinese input method to type some Chinese characters in the Online shell, there is nothing to dispaly.
![微信截图_20241108161722](https://github.com/user-attachments/assets/2525934f-a916-4c88-8a18-07cf609ad232)

![微信截图_20241108161731](https://github.com/user-attachments/assets/08f04191-e62c-43f7-911b-22ae22b6e283)
I am afraid that I might cause trouble for you, so it's okay if this bug isn't addressed. I will try my best to use English in the Pybricks IDE. Thank you again for your help.

**Comment by BertLindeman** [comment, marked as answer]
Hello,

To run this test I installed on beta.pybricks.com the firmware that comes along with the beta.
Remember it **is** a **beta** so things might change until it gets a stable version.

I ran this program on that firmware:

```python:
from pybricks import version

print("此集线器正在运行版本:\n\t", version)
print("举头望明月，")
print("低头思故乡。")
print("窗前明月光，疑是地上霜")
print("举头望明月，低头思故乡")
print("我非常喜欢PyBricks！")
```

I believe the first command is `This hub is running version` got it from Google translate 😄 
The result is:
```
此集线器正在运行版本:
         ('primehub', '3.6.0b2', 'ci-release-81-v3.6.0b2 on 2024-10-15')
举头望明月，
低头思故乡。
窗前明月光，疑是地上霜
举头望明月，低头思故乡
我非常喜欢PyBricks！
```

Bert

**Reply by angzicn** [reply]
Your help is immensely appreciated.
Thank you very much!

**Comment by laurensvalk** [comment]
Indeed, #1742 has fixed the line breaking off mid-character, but the terminal does indeed not appear to accept unicode input, even after enabling https://github.com/pybricks/pybricks-micropython/pull/271:

https://github.com/user-attachments/assets/1265f2ef-670c-42bc-a95f-0d988d86f40d

**Reply by BertLindeman** [reply]
Laurens, I tried to mimic your program.
And copied some text to enter in the terminal, but paste on the terminal seems to do nothing.
Neither in the Google Chrome app or Google Chrome. [EDIT] On windows 11.
I have no plans to reconfigure my keyboard to Chinese. 😄 

Copy / paste Chinese text in the source window is fine.

**Reply by angzicn** [reply]
Thank you very much for your attention to the use of Chinese on Pybricks.

**Reply by dlech** [reply]
This is something that isn't supported in upstream MicroPython. See https://github.com/micropython/micropython/issues/7585

But we might be able to cherry-pick the implementation from CircuitPython linked in that issue.

**Reply by dlech** [reply]
As a workaround for now, it should be possible to use "paste mode" in the REPL by pressing <kbd>CTRL</kbd>+<kbd>E</kbd>.

**Comment by dlech** [comment]
Another minor issue is that on hubs with the `ujson` module enabled, multibyte characters are escaped with `\u` in `print(repr(...))` instead of printing actually printable characters.  This can be problematic in the REPL if you want to see what the value of a variable is since it implicitly does `print(repr(...))` for this.

EDIT: if `ujson` is not enabled, it probably just uses `\x` instead of  `\u`.
